builtin/mainmenu/tab_online.lua

```diff
@@ -218,13 +218,19 @@ local function search_server_list(input)
 				found = found + count * 4
 			end
 
-			if server.description then
-				local desc = server.description:lower()
-				local _, count = desc:gsub(keyword, keyword)
-				found = found + count * 2
+            if server.description then
+                local desc = server.description:lower()
+                local _, count = desc:gsub(keyword, keyword)
+                found = found + count * 2
+            end
+        end
+		-- Make search precise for exactly matched server
+        -- Or at least more exactly matched server should be rated higher at the match scoring 
+        local _, count = server.name:lower():gsub(input:lower(), input:lower())
```

Comment:
this is addressing a specific case but doesn't change the matching from OR to AND
(as ways suggested in the original issue)

Reply:
This is also pretty hacky and subtly broken (it interprets the input as a pattern). Since you're just testing `count > 0`, you should write this as `if server.name:lower():find(input:lower(), 1, true) then ... end`.